File: 05. Two-Pointers/python/sortedArray.py
# Problem Description:

# Check if an array is sorted or not.

# Input:
# First line contains an integer representing the size of array.

# Second line contains 'n' numbers 

# Output: 
# Yes (or) No

def sortedArray(arr):

    firstPtr = 0
    secondPtr = 1

    while(secondPtr < len(arr)):
        if(arr[firstPtr] > arr[secondPtr]):
            return False
        
        firstPtr += 1
        secondPtr += 1
    
    return True

arr = list(map(int, input("Enter array: ").split()));
print(sortedArray(arr));

# Time Complexity  : O(n)
# Space Complexity : O(1)

File: 05. Two-Pointers/javascript/sortedArray.js
// Problem Description:

// Check if an array is sorted or not.

// Input:
// First line contains an integer representing the size of array.

// Second line contains 'n' numbers 

// Output: 
// Yes (or) No

function sortedArray(arr) {

    let pointer1 = 0;
    let pointer2 = 1;

    while(pointer2 < arr.length) {

        if(arr[pointer1] > arr[[pointer2]]) {
            return false;
        }

        pointer1++;
        pointer2++;

    }

    return true;

}

/*
    Time Complexity  : O(n)
    Space Complexity : O(1)
*/